microprofile/server/src/main/java/io/helidon/microprofile/server/ServerImpl.java

```diff
@@ -166,30 +179,157 @@ public Response toResponse(Exception exception) {
+
+        server = serverBuilder.config(serverConfiguration)
+                .build();
 
+        STARTUP_LOGGER.finest("Server created");
+    }
+
+    private static void registerJerseyApplications(Config config,
+                                                   Routing.Builder routingBuilder,
+                                                   Map<String, Routing.Builder> namedRoutings,
+                                                   ServerConfiguration serverConfiguration,
+                                                   List<JaxRsApplication> applications,
+                                                   Supplier<ExecutorService> defaultExecService) {
         applications
                 .forEach(app -> {
                     JerseySupport js = JerseySupport.builder(app.resourceConfig())
                             .executorService(app.executorService()
-                                                     .orElseGet(builder::defaultExecutorService))
+                                                     .orElseGet(defaultExecService))
                             .build();
 
-                    if ("/".equals(app.contextRoot())) {
-                        routingBuilder.register(js);
-                    } else {
-                        routingBuilder.register(app.contextRoot(), js);
-                    }
+                    Routing.Rules routing = findRouting(config,
+                                                        serverConfiguration,
+                                                        routingBuilder,
+                                                        namedRoutings,
+                                                        app.routingName(),
+                                                        app.routingNameRequired(),
+                                                        app.appClassName());
+
+                    registerService(config,
+                                    routing,
+                                    app.contextRoot(),
+                                    app.appClassName(),
+                                    js);
                 });
+    }
 
-        STARTUP_LOGGER.finest("Registered jersey application(s)");
+    @SuppressWarnings("unchecked")
+    private static void registerWebServerServices(Config config,
+                                                  BeanManager beanManager,
+                                                  Routing.Builder routingBuilder,
+                                                  Map<String, Routing.Builder> namedRoutings,
+                                                  ServerConfiguration serverConfiguration) {
+
+        CreationalContext<Object> context = beanManager.createCreationalContext(null);
+        List<Bean<?>> wsServicesSorted = prioritySort(beanManager.getBeans(Service.class));
+
+        wsServicesSorted.forEach(serviceBean -> {
+            Bean<Object> theBean = (Bean<Object>) serviceBean;
```

Comment:
I am not much of a fan of making big lambda bodies. What do you think about making separate method for this?